docs/other-topics/hooks.md

```diff
@@ -2,404 +2,254 @@
+// Both of these will remove the hook:
+// success-next-line
+Book.hooks.removeListener('afterCreate', myListener);
+// success-next-line
+Book.hooks.removeListener('afterCreate', 'yourHookIdentifier');
 ```
 
-### Model hooks
-
-Sometimes you'll be editing more than one record at a time by using methods like `bulkCreate`, `update` and `destroy`. The following hooks will emit whenever you're using one of those methods:
+:::caution
 
-* `YourModel.beforeBulkCreate(callback)`
-  * The callback has the form `(instances, options) => /* ... */`
-* `YourModel.beforeBulkUpdate(callback)`
-  * The callback has the form `(options) => /* ... */`
-* `YourModel.beforeBulkDestroy(callback)`
-  * The callback has the form `(options) => /* ... */`
-* `YourModel.afterBulkCreate(callback)`
-  * The callback has the form `(instances, options) => /* ... */`
-* `YourModel.afterBulkUpdate(callback)`
-  * The callback has the form `(options) => /* ... */`
-* `YourModel.afterBulkDestroy(callback)`
-  * The callback has the form `(options) => /* ... */`
+Hooks added by decorators cannot be removed using the callback instance, but can still be removed if they are named:
 
-Note: methods like `bulkCreate` do not emit individual hooks by default - only the bulk hooks. However, if you want individual hooks to be emitted as well, you can pass the `{ individualHooks: true }` option to the query call. However, this can drastically impact performance, depending on the number of records involved (since, among other things, all instances will be loaded into memory). Examples:
-
-```js
-await Model.destroy({
-  where: { accessLevel: 0 },
-  individualHooks: true
-});
-// This will select all records that are about to be deleted and emit `beforeDestroy` and `afterDestroy` on each instance.
+```typescript
+import { Sequelize, Model, Hook } from '@sequelize/core';
+import { BeforeFind } from 'sequelize-typescript';
 
-await Model.update({ username: 'Tony' }, {
-  where: { accessLevel: 0 },
-  individualHooks: true
-});
-// This will select all records that are about to be updated and emit `beforeUpdate` and `afterUpdate` on each instance.
-```
-
-If you use `Model.bulkCreate(...)` with the `updateOnDuplicate` option, changes made in the hook to fields that aren't given in the `updateOnDuplicate` array will not be persisted to the database. However it is possible to change the `updateOnDuplicate` option inside the hook if this is what you want.
-
-```js
-User.beforeBulkCreate((users, options) => {
-  for (const user of users) {
-    if (user.isMember) {
-      user.memberSince = new Date();
-    }
-  }
-
-  // Add `memberSince` to updateOnDuplicate otherwise it won't be persisted
-  if (options.updateOnDuplicate && !options.updateOnDuplicate.includes('memberSince')) {
-    options.updateOnDuplicate.push('memberSince');
+export class MyModel extends Model {
+  @BeforeFind({ name: 'yourHookIdentifier' })
+  static logFindAll() {
```

Comment:
Another nitpicky consistency one; above we call this `logFind()`. Obviously these are two different examples so it's not wrong but might be nice to keep them consistent